### PR TITLE
Add forward torch.cat support to torch_nntile

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Adds forward-only `torch.cat` / `aten.cat` support on the `device="nntile"` backend. The implementation records chained `tensor::concat` operations into `TensorGraph` (no new libnntile kernels).

## Changes

- **`torch_nntile/csrc/nntile_cat.cpp`**: ATen registration for `cat` and `cat.out` on `PrivateUse1`
- **`torch_nntile/csrc/nntile_executor.{h,cpp}`**: `tensor_cat_fp32` bridges PyTorch tensors to N-way concat via left-folded binary `nntile::tensor::concat`
- **`torch_nntile/tests/test_cat_parity.py`**: CPU vs nntile parity tests (1D/2D/3D, negative dim, many tensors, single-tensor no-op, `out=` variant, graph mode)
- **`torch_nntile/README.md`**: Documents the new supported op

## Constraints (phase 2)

- `float32` contiguous tensors on `device="nntile"` only
- Backward (`CatBackward`) is **not** implemented in this PR

## Testing

```bash
export LD_LIBRARY_PATH=$PWD/build/nntile:/opt/starpu/lib
export STARPU_SILENT=1 STARPU_FXT_TRACE=0 STARPU_WORKERS_NOBIND=1
pytest -vv torch_nntile/tests/test_cat_parity.py
```

All 9 tests pass locally.

<div><a href="https://cursor.com/agents/bc-e7c490b0-8fd0-498b-8acd-e63990e8fd8e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-e7c490b0-8fd0-498b-8acd-e63990e8fd8e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
